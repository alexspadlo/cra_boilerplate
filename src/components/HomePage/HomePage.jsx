import React from 'react'
import { logIn, /*getAllUsers*/ } from '../../utils'
import s from './HomePage.module.scss'

class HomePage extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            name: '',
            email: ''
        }
        this.logtheFakeUser = this.logtheFakeUser.bind(this)
        //this.getFakeUser = this.getFakeUser.bind(this)
    }

    logtheFakeUser() {
        this.setState({
            name: 'Alex',
            email: 'yupa@gmail.com'
        }, () => {
            const { name, email } = this.state
            if (name && email) {
                logIn({ name, email })
            }
        })
    }

    // using async function

    // getFakeUser = async () => {
    //     const allUsers = await getAllUsers()
    //     if (allUsers) {
    //         this.setState({
    //             sukces: true
    //         })
    //     }
    // }

    render() {
        return (
            <div className="container p-0">
                <div className="row p-5">
                    <div className="col text-center m-auto">
                        <h1 className="text-success pb-4"><strong>HEY !</strong></h1>
                        <button onClick={() => this.logtheFakeUser()} className={`text-center ${s.outButton}`}>Log in</button>
                    </div>
                </div>
            </div>

        )
    }
}

export default HomePage
