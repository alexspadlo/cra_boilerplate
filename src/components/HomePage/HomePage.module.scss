.outButton {
  padding: 5px 20px;
  color: black;
  background-color: rgb(154, 230, 158);
  border: none;
  border-radius: 10px;

  &:focus {
    outline: 0;
  }

  &:hover {
    color: white;
    background-color: rgb(236, 93, 11);
  }
}
