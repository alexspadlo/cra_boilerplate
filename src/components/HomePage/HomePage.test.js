import React from 'react'
import { render } from '@testing-library/react'
import HomePage from './HomePage'

describe('Testing HomePage Component', () => {
    test('Do not render without the props', () => {
        const { container, getByText } = render(<HomePage />)
        expect(getByText('HEY !')).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot()
    })
})