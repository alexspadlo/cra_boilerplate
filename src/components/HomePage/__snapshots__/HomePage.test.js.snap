// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomePage Component Do not render without the props 1`] = `
<div
  class="container p-0"
>
  <div
    class="row p-5"
  >
    <div
      class="col text-center m-auto"
    >
      <h1
        class="text-success pb-4"
      >
        <strong>
          HEY !
        </strong>
      </h1>
      <button
        class="text-center outButton"
      >
        Log in
      </button>
    </div>
  </div>
</div>
`;
